task/keyaction.c

```diff
@@ -427,6 +427,10 @@ void KeypressAction(uint8_t Action) {
 				MENU_DrawSetting();
 				break;
 
+			case ACTION_BANDWIDTH:
+				gVfoState[gSettings.CurrentVfo].bIsNarrow ^= 1;
+				UI_DrawVfo(gSettings.CurrentVfo);
```

Comment:
We could also save the channel to make the bandwidth change persistent.
```suggestion
				UI_DrawVfo(gSettings.CurrentVfo);
				CHANNELS_SaveVfo();
```

Reply:
Hi chaps not sure if you want to go down this route but I've added a new dialog in my fw version (plus for FM but thats not quite the same).

ui/dialog.c
	case DIALOG_BAND_WIDTH:
		if (bSet) {
			UI_DrawString(10, 48, "BW 12.5 kHz ", 12);
		} else {
			UI_DrawString(10, 48, "BW 25.0 kHz ", 12);
		}
		break;

ui/dialog.h
DIALOG_BAND_WIDTH = 3,

task/keyaction.c
UI_DrawDialogText(DIALOG_BAND_WIDTH, gVfoState[gSettings.CurrentVfo].bIsNarrow);



^^^ mine differs as I've changed quite a bit, ya'll know where to put them 👍